### PR TITLE
[PR for MIRROR PR] Changes for 16248

### PR DESCRIPTION
**Pull request by Tastyfish**
## About The Pull Request

Note that this is a sub-PR for the #16248 branch.

Merges changes to the upstream subsystems, including updating how add_startup_message() is called to match new code flow.

Also updates SR subsystems.
- Note that Goldeneye and Liquid Turfs did not implement initialize() and appear to not benefit from an SS init occuring (ie they don't take advantage of the `initialized` var). Thus, they are now flagged with `SS_NO_INIT`. Liquid turfs tested to run fine. I double checked goldeneye code for no usage of initialized, and am fairly confident it's fine. Way too much of a pain to actually test the goldeneye game mode locally though.

Removed Lobby Eye subsystem as it's just burning CPU and network traffic for an effect we can't see with the current title screen implementation.

## How This Contributes To The Skyrat Roleplay Experience

upstream merge

## Changelog

changelog bot doesn't run for non-master PR's to my knowledge

### CODE REVIEW

File: modular_skyrat/modules/autotransfer/code/autotransfer.dm [hunk 1]
```diff
@@ -11,16 +11,16 @@ SUBSYSTEM_DEF(autotransfer)
 	var/maxvotes
 	var/curvotes = 0
 
-/datum/controller/subsystem/autotransfer/Initialize(timeofday)
+/datum/controller/subsystem/autotransfer/Initialize()
 	if(!CONFIG_GET(flag/autotransfer)) //Autotransfer voting disabled.
 		can_fire = FALSE
-		return ..()
+		return SS_INIT_NO_NEED
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/autotransfer/code/autotransfer.dm:17]:
```suggestion
		return SS_INIT_NO_NEED

```

File: modular_skyrat/modules/decay_subsystem/code/decaySS.dm [hunk 2]
```diff
@@ -37,12 +37,12 @@ SUBSYSTEM_DEF(decay)
 	if(CONFIG_GET(flag/ssdecay_disabled))
 		message_admins("SSDecay was disabled in config.")
 		log_world("SSDecay was disabled in config.")
-		return ..()
+		return SS_INIT_NO_NEED
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/decay_subsystem/code/decaySS.dm:40]:
```suggestion
		return SS_INIT_NO_NEED

```

File: modular_skyrat/modules/decay_subsystem/code/decaySS.dm [hunk 3]
```diff
@@ -37,12 +37,12 @@ SUBSYSTEM_DEF(decay)
...
 	// Putting this first so that it just doesn't waste time iterating through everything if it's not going to do anything anyway.
 	if(prob(50))
 		message_admins("SSDecay will not interact with this round.")
 		log_world("SSDecay will not interact with this round.")
-		return ..()
+		return SS_INIT_NO_NEED
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/decay_subsystem/code/decaySS.dm:46]:
```suggestion
		return SS_INIT_NO_NEED

```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I'd still leave the lobby camera in, just so it's removed cleanly with an atomized PR rather than getting lost in the history of a mirror PR.

**Comment by Tastyfish** [comment]
> I'd still leave the lobby camera in, just so it's removed cleanly with an atomized PR rather than getting lost in the history of a mirror PR.

Restored.

**Pull request merged**